lib/spack/docs/getting_started.rst

```diff
@@ -57,3 +112,928 @@ This will install a new spack script in ``/my/favorite/prefix/bin``,
+
+.. code-block:: yaml
+
+   compilers:
+   - compiler:
+       modules = []
+       operating_system: centos6
+       paths:
+         cc: /usr/local/bin/icc-15.0.024-beta
+         cxx: /usr/local/bin/icpc-15.0.024-beta
+         f77: /usr/local/bin/ifort-15.0.024-beta
+         fc: /usr/local/bin/ifort-15.0.024-beta
+       spec: intel@15.0.0:
+
+For compilers, like ``clang``, that do not support Fortran, put
+``None`` for ``f77`` and ``fc``:
+
+.. code-block:: yaml
+
+       paths:
+         cc: /usr/bin/clang
+         cxx: /usr/bin/clang++
+         f77: None
+         fc: None
+       spec: clang@3.3svn:
+
+Once you save the file, the configured compilers will show up in the
+list displayed by ``spack compilers``.
+
+You can also add compiler flags to manually configured compilers. The
+valid flags are ``cflags``, ``cxxflags``, ``fflags``, ``cppflags``,
+``ldflags``, and ``ldlibs``. For example:
+
+.. code-block:: yaml
+
+   compilers:
+   - compiler:
+       modules = []
+       operating_system: OS
+       paths:
+         cc: /usr/local/bin/icc-15.0.024-beta
+         cxx: /usr/local/bin/icpc-15.0.024-beta
+         f77: /usr/local/bin/ifort-15.0.024-beta
+         fc: /usr/local/bin/ifort-15.0.024-beta
+       parameters:
+         cppflags: -O3 -fPIC
+       spec: intel@15.0.0:
+
+These flags will be treated by spack as if they were enterred from
+the command line each time this compiler is used. The compiler wrappers
+then inject those flags into the compiler command. Compiler flags
+enterred from the command line will be discussed in more detail in the
+following section.
+
+^^^^^^^^^^^^^^^^^^^^^^^
+Build Your Own Compiler
+^^^^^^^^^^^^^^^^^^^^^^^
+
+If you are particular about which compiler/version you use, you might
+wish to have Spack build it for you.  For example:
+
+.. code-block::
+
+    spack install gcc@4.9.3
+
+Once that has finished, you will need to add it to your
+``compilers.yaml`` file.  If this is your preferred compiler, in
+general future Spack builds will use it.
+
+.. note::
+
+    If you are building your own compiler, it can be useful to have a
+    Spack instance just for that.  For example, create a new Spack in
+    ``~/spack-tools`` and then run ``~/spack-tools/bin/spack install
+    gcc@4.9.3``.  Once the compiler is built, don't build anything
+    more in that Spack instance; instead, create a new "real" Spack
+    instance, configure Spack to use the compiler you've just built,
+    and then build your application software in the new Spack
+    instance.
+
+    This tip is useful because sometimes you will find yourself
+    rebuilding may pacakges due to Spack updates.  Sometimes, you
+    might even delete your entire Spack installation and start fresh.
+    If your compiler was built in a separate Spack installation, you
+    will never have to rebuild it --- as long as you wish to continue
+    using that version of the compiler.
+
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Compilers Requiring Modules
```

Comment:
Good point.  However, Spack does not currently have a mechanism to set env
vars for compilers, but it does have a way to load modules.

On Wed, Sep 7, 2016 at 3:59 PM, Adam J. Stewart <notifications@github.com>
wrote:

> In lib/spack/docs/getting_started.rst:
>
> > +    gcc@4.9.3``.  Once the compiler is built, don't build anything
> > +    more in that Spack instance; instead, create a new "real" Spack
> > +    instance, configure Spack to use the compiler you've just built,
> > +    and then build your application software in the new Spack
> > +    instance.
> > +
> > +    This tip is useful because sometimes you will find yourself
> > +    rebuilding may pacakges due to Spack updates.  Sometimes, you
> > +    might even delete your entire Spack installation and start fresh.
> > +    If your compiler was built in a separate Spack installation, you
> > +    will never have to rebuild it --- as long as you wish to continue
> > +    using that version of the compiler.
> > +
> > +
> > +^^^^^^^^^^^^^^^^^^^^^^^^^^^
> > +Compilers Requiring Modules
>
> I would call this section "Compilers Requiring Environment Variables"
> instead of "Compilers Requiring Modules" since not all users use Modules to
> manage their environment variables.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/1676/files/c9e83c62a002c42852fcb54a3b45d54a68cb8ccc#r77892706>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AB1cd0zt00z1KJvGAtiRt8t064co9571ks5qnxeegaJpZM4JxOaa>
> .
>